plugins/de.cau.cs.kieler.klighd/src/de/cau/cs/kieler/klighd/filtering/SemanticFilterRuleUtil.java

```diff
@@ -0,0 +1,126 @@
+     * always false.
+     * 
+     * @param ruleName
+     *            The rule name.
+     * @param operands
+     *            The operands.
+     * @return The SemanticFilterRule.
+     */
+    public static SemanticFilterRule getBigOrConnective(String ruleName,
+            SemanticFilterRule... operands) {
+        if (operands.length <= 0) {
+            // No rules, automatically evaluate to false
+            return FALSE;
+        } else if (operands.length <= 1) {
+            // Just one rule, true iff rule is true
+            // Add ruleName instead of just using first operand twice to avoid double evaluation
+            return addRuleName(ruleName, operands[0]);
+        }
+        OrConnective bigOr = new OrConnective(operands[0], operands[1], ruleName);
+        for (int i = 2; i < operands.length; i++) {
+            bigOr.rightOperand = new OrConnective(bigOr.rightOperand, operands[i]);
+        }
+        return bigOr;
+    }
+
+    /**
+     * Constructs an unnamed rule with all operands connected via AND. If no operands are given,
+     * this is always false.
+     * 
+     * @param operands
+     *            The operands.
+     * @return The SemanticFilterRule.
+     */
+    public static SemanticFilterRule getBigAndConnective(SemanticFilterRule... operands) {
+        return getBigAndConnective(null, operands);
+    }
+
+    /**
+     * Constructs a named rule with all operands connected via AND. If no operands are given, this
+     * is always false.
+     * 
+     * @param ruleName
+     *            The rule name.
+     * @param operands
+     *            The operands.
+     * @return The SemanticFilterRule.
+     */
+    public static SemanticFilterRule getBigAndConnective(String ruleName,
+            SemanticFilterRule... operands) {
+        if (operands.length <= 0) {
+            // No rules, automatically evaluate to false
+            return FALSE;
+        } else if (operands.length <= 1) {
+            // Just one rule, true iff rule is true
+            // Add ruleName instead of just using first operand twice to avoid double evaluation
+            return addRuleName(ruleName, operands[0]);
+        }
+        // Use first operand twice in case of just one rule, e.g. true iff rule is true
+        AndConnective bigAnd = new AndConnective(operands[0], operands[1], ruleName);
+        for (int i = 2; i < operands.length; i++) {
+            bigAnd.rightOperand = new AndConnective(bigAnd.rightOperand, operands[i]);
+        }
+        return bigAnd;
+    }
+
+    /**
+     * Constructs a named rule from a given potentially unnamed rule.
+     * 
+     * @param ruleName
+     *            The rule name.
+     * @param rule
+     *            The rule.
+     * @return A named rule.
+     */
+    public static SemanticFilterRule addRuleName(String ruleName, SemanticFilterRule rule) {
+        return new NegationConnective(new NegationConnective(rule), ruleName);
```

Comment:
I don't think this should be the way to do it, there should rather be a setter for rules that handles this directly.

Reply:
in that case, there should just be an identity connective which evaluates to whatever its child evaluates to